docs/product/requirements/2025/sso-oidc.md

```diff
@@ -198,40 +121,61 @@ Grace (customer success team)
 
 * As Grace, I want to access live customer data in Mathesar by signing in with my usual company account so that I can self-serve metrics without waiting on the data team.
 
-## Out of Scope
+## Success Criteria
 
-This is explicitly an MVP, so the following things are out of scope:
+How we'll know we've succeeded with our goals for adding SSO:
 
-Authentication Methods:
+* We should transition [internal.mathesar.org](http://internal.mathesar.org) to this login style and not hate it.  
+* We track SSO usage growing in analytics.  
+* We see an uptick in conversion from our website (after we advertise SSO)  
+* We see qualitative positive user feedback on GitHub / Reddit, etc.
 
-* No SAML support (OIDC only)  
-* No support for IdP-initiated login flows (SP-initiated only)  
-* No social login (e.g. GitHub, Facebook, etc.)  
-* No password-based SSO fallback except for Mathesar admins.
+## Requirements
 
-Provisioning and Role Management:
+### Minimal Scope
 
-* No SCIM or automatic deprovisioning  
-* No UI for managing users, roles, or SSO config  
-* No per-user or per-session role overrides  
-* No multi-role assignment per user  
-* No support for assigning roles via UI or API based on group claims (config file only)
+Setup:
 
-Access Control:
+* Configure OIDC (Google Workspace / Okta / Auth0, etc) via config file.  
+* Restrict access to a specific email domain in config.  
+* Manually pre-create accounts and link to OIDC identities, at minimum.  
+* Define a default Postgres role for all SSO users in the config file.
 
-* No support for multi-tenancy or per-project IdP config  
-* No other permissions mapping beyond Postgres roles  
-* No email-based whitelists beyond domain restriction  
-* No support for conditional logic on access (e.g. “this group *and* this claim”)
+Maintenance:
 
-Observability and Admin Tools:
+* Retain fallback login method in case SSO is misconfigured.  
+* Get clear error messaging for others if login fails.  
+* See clear error if a user is unauthorized based on domain or mapping.
 
-* No UI for viewing login attempts, group mappings, or audit trails  
-* No real-time SSO diagnostics or health checks  
-* No alerting for failed logins or misconfigurations
+End user experience:
 
-User Experience:
+* Users see a “Sign in with Google” / “Sign in with Okta” option on the login page.  
+* Sign in and immediately access the right data without manual setup.  
```

Comment:
Aside from normal access post-sign-in, are there any other implications to this bullet point? I'm not sure what it's for.

Reply:
The aim of this bullet is to document that from the end user's perspective, they just sign into Mathesar and have access to the things they need without any additional steps like e.g. having to request access, or having to do some sort of onboarding. It's "obvious" in the sense that we know how Mathesar works and that there's no reason to do any of that, but it still seemed good to include so that the full experience of the user was represented.